lib/util/parse.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+/**
+ * Parses a script or style object for data attributes and correctly formats them.
+ * @param {Object} obj An object.
+ * @returns {Object} The parsed object.
+ */
+const parseDataAttributes = (obj) => {
+
+    // Don't mutate the original object.
+    const newObj = Object.assign({}, obj);
+
+    if (!newObj.dataAttributes) {
+
+        newObj.dataAttributes = {};
+
+        return newObj;
+
+    }
+
+    const dataAttributes = Object.keys(newObj.dataAttributes);
```

Comment:
@allanchau, how come you're not using `map` here directly on `newObj.dataAttributes`?

Reply:
@smebberson oh sorry, I might have accidentally closed the other convo. If we use map we would have to convert the array to an object later on anyway, seems like that would be a bit more complex than what I've done here?

Reply:
@allanchau, yeah, fair enough, makes sense.